pkg/istio-agent/sds-agent.go

```diff
@@ -306,9 +319,16 @@ func (conf *SDSAgent) Start(isSidecar bool, podNamespace string) (*sds.Server, e
 			if err != nil {
 				log.Fatalf("Failed to write certs: %v", err)
 			}
+			rootCert = sir.RootCert
 		}
 	}
 
+	if privateKey != nil && certChain != nil && rootCert != nil {
```

Comment:
I thought we already had some code doing this @costinm or was that not implemented?

Reply:
As far as Lei and I see, the work of dumping key/certs here:
https://github.com/istio/istio/blob/93a7b582e8a8248bdf00d1b7ee2ad27249900ec4/pkg/istio-agent/sds-agent.go#L291
https://github.com/istio/istio/blob/93a7b582e8a8248bdf00d1b7ee2ad27249900ec4/pkg/istio-agent/sds-agent.go#L295
https://github.com/istio/istio/blob/93a7b582e8a8248bdf00d1b7ee2ad27249900ec4/pkg/istio-agent/sds-agent.go#L315
only dumps them for the first time, the files are not updated when rotated, since sdsserice.go isn't changed.

So we should remove these lines (in another PR). And if users want to debug, they should turn on ENV OUTPUT_KEY_CERT_TO_DIRECTORY instead of looking at /etc/istio/proxy certs.